File: extract_colors.py
from PIL import Image
import sys

try:
    img = Image.open(sys.argv[1])
    img = img.convert("RGBA")
    colors = img.getcolors(maxcolors=1000000)
    
    # Filter out transparent pixels
    solid_colors = [c for c in colors if c[1][3] > 0]
    
    # Sort by frequency
    solid_colors.sort(key=lambda x: x[0], reverse=True)
    
    print("Top 10 colors (Frequency, RGBA):")
    for count, color in solid_colors[:10]:
        hex_color = "#{:02x}{:02x}{:02x}".format(color[0], color[1], color[2])
        print(f"{count:5d} pixels - {color} -> {hex_color}")
except Exception as e:
    print("Error:", e)

File: extract_colors.js
const { Jimp } = require('jimp');

async function extract() {
  const image = await Jimp.read('Meli_Facil-removebg-preview.png');
  const colorCounts = {};
  
  image.scan(0, 0, image.bitmap.width, image.bitmap.height, function(x, y, idx) {
    const r = this.bitmap.data[idx + 0];
    const g = this.bitmap.data[idx + 1];
    const b = this.bitmap.data[idx + 2];
    const a = this.bitmap.data[idx + 3];
    
    if (a > 50) { // ignoring very transparent pixels
      const hex = '#' + ((1 << 24) + (r << 16) + (g << 8) + b).toString(16).slice(1);
      colorCounts[hex] = (colorCounts[hex] || 0) + 1;
    }
  });

  const sorted = Object.entries(colorCounts).sort((a, b) => b[1] - a[1]);
  console.log('Top 10 Colors:');
  sorted.slice(0, 10).forEach(([color, count]) => {
    console.log(`${count} pixels: ${color}`);
  });
}

extract().catch(console.error);
